pkg/iostreams/color.go

```diff
@@ -34,27 +34,33 @@ func EnvColorForced() bool {
 }
 
 func Is256ColorSupported() bool {
+	return IsTrueColorSupported() ||
+		strings.Contains(os.Getenv("TERM"), "256") ||
+		strings.Contains(os.Getenv("COLORTERM"), "256")
+}
+
+func IsTrueColorSupported() bool {
 	term := os.Getenv("TERM")
 	colorterm := os.Getenv("COLORTERM")
 
-	return strings.Contains(term, "256") ||
-		strings.Contains(term, "24bit") ||
+	return strings.Contains(term, "24bit") ||
 		strings.Contains(term, "truecolor") ||
-		strings.Contains(colorterm, "256") ||
 		strings.Contains(colorterm, "24bit") ||
 		strings.Contains(colorterm, "truecolor")
 }
```

Comment:
Is this necessary? While Windows Terminal with WSL2 defines `$TERM` as "xterm-256color", it was showing colors just fine. Whether it's actually 256-color or true-color I'm unsure - there's been various discussions about this on https://github.com/microsoft/terminal - but it, and I wonder about other terminals, seems to normalize colors to the nearest match which, for a CLI, seems perfectly fine.

Reply:
Comparing colors, "bug" actually seems more red than the foreground or background in the browser, leading me to believe that - Windows Terminal, at least - colors are approximated:

![image](https://user-images.githubusercontent.com/1532486/129944568-bae7e15f-43d9-4729-87e6-d424c290fd2d.png)

That said, it's not like I can't update my .profile to redefine `TERM` as "xterm-truecolor" or something, but I wonder how many more customers will miss out on changes like this when it may, in fact, work fine.

Reply:
If you guys want to detect WSL then you can look at this code: https://github.com/Delta456/box-cli-maker/blob/master/detect_unix.go#L27-L39

Reply:
Thanks, but there's really no need to detect WSL, but the console host in which it's running. My point here was that, when we were detecting `WT_SESSION` (which has been removed), the environment variable may also be defined for WSL/WSL2 when running in Windows Terminal since it injected the environment variable into any shell it hosted.

The main goal was to detect 256color and truecolor support, which based on @DHowett's recommendations, is in the latter commits I pushed.

Reply:
Alright!